### PR TITLE
accept SVMPointer in _get_data_dedup_cache_key

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: pytato/transform/__init__.py [hunk 1]
```diff
@@ -1724,10 +1724,17 @@ def _get_data_dedup_cache_key(ary: DataInterface) -> Hashable:
     if "pyopencl" in sys.modules:
         from pyopencl.array import Array as CLArray  # type: ignore[import]
         from pyopencl import MemoryObjectHolder
+        try:
+            from pyopencl import SVMPointer
+            allowed_base_data_classes = tuple((SVMPointer, MemoryObjectHolder))
+        except ImportError:
+            allowed_base_data_classes = tuple((MemoryObjectHolder,))
```
Comment (inducer) [inline pytato/transform/__init__.py:1731]:
```suggestion
            allowed_base_data_classes = (SVMPointer, MemoryObjectHolder)
        except ImportError:
            allowed_base_data_classes = (MemoryObjectHolder,)
```

Comment (matthiasdiener) [reply pytato/transform/__init__.py:1731]: mypy doesn't like that:

```
pytato/transform/__init__.py:1731: error: Incompatible types in assignment (expression has type "Tuple[Any]", variable has type "Tuple[Any, Any]")  [assignment]
```

Comment (inducer) [reply pytato/transform/__init__.py:1731]: You need to annotate one as `: Tuple[Type, ...]`

**Comment by inducer** [comment]
Thanks!

**Pull request merged**